src/range.cr

```diff
@@ -87,6 +87,10 @@ struct Range(B, E)
   def initialize(@begin : B, @end : E, @exclusive : Bool = false)
   end
 
+  def ==(other : Range)
+    @begin == other.@begin && @end == other.@end && @exclusive == other.@exclusive
+  end
```

Comment:
```suggestion
  def_equals @begin, @end, @exclusive
```

Reply:
`def_equals` has the same `self` restriction as `Struct#==` that we want to avoid.

Reply:
Oh right, forgot about that 🤦